src/components/MissingLabel/MissingLabel.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { Tooltip, TooltipPosition } from '@patternfly/react-core';
+import { icons, serverConfig } from '../../config';
+import { KialiIcon } from '../../config/KialiIcon';
+import { style } from 'typestyle';
+import { PFBadge } from '../Pf/PfBadges';
+
+type MissingLabelProps = {
+  missingApp: boolean;
+  missingVersion: boolean;
+  tooltip: boolean;
+  style?: React.CSSProperties;
+};
+
+const infoStyle = style({
+  margin: '0px 5px 2px 4px',
+  verticalAlign: '-5px !important'
+});
+
+class MissingLabel extends React.Component<MissingLabelProps, {}> {
+  render() {
+    const appLabel = serverConfig.istioLabels.appLabelName;
+    const versionLabel = serverConfig.istioLabels.versionLabelName;
+    const icon = icons.istio.missingLabel.icon;
+    const color = icons.istio.missingLabel.color;
+    const tooltipContent = (
+      <div style={{ textAlign: 'left' }}>
+        {this.props.missingApp && (
+          <div>
+            <div>
+              <PFBadge badge={{ badge: appLabel }} isRead={true} style={{ marginRight: '0px' }} /> label is missing.{' '}
+              <br />
+            </div>
+            <div>This workload won't be linked with an application.</div>
+          </div>
+        )}
```

Comment:
The outer div can be replaced, but not the inner ones when those are part of the layout (i.e. block vs inline) otherwise you need to use other markup (br or p).

Reply:
mmm ok I just tried locally and the render was identical without the inner ones maybe it was cached